backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropCallConvertors.kt

```diff
@@ -197,6 +197,15 @@ private fun InteropCallContext.convertCPointerToNativePtr(cPointer: IrExpression
 private fun InteropCallContext.writePointerToMemory(nativePtr: IrExpression, value: IrExpression): IrExpression {
     val valueToWrite = when {
         value.type.isCPointer() -> convertCPointerToNativePtr(value)
+        // Special case of null value
```

Comment:
This is the reason to avoid relying on actual type (`value.type`) and use expected type (`accessor.valueParameters[0].type`) instead.